pkgs/development/haskell-modules/non-hackage-packages.nix

```
@@ -26,14 +26,14 @@ self: super: {
 
   nix-output-monitor = self.callPackage ../../tools/nix/nix-output-monitor { };
 
-  # cabal2nix --revision <rev> https://github.com/hasura/ci-info-hs.git
-  ci-info = self.callPackage ../misc/haskell/hasura/ci-info {};
-  # cabal2nix --revision <rev> https://github.com/hasura/pg-client-hs.git
-  pg-client = self.callPackage ../misc/haskell/hasura/pg-client {};
-  # cabal2nix --revision <rev> https://github.com/hasura/graphql-parser-hs.git
-  graphql-parser = self.callPackage ../misc/haskell/hasura/graphql-parser {};
-  # cabal2nix  --subpath server --maintainer offline --no-check --revision 1.2.1 https://github.com/hasura/graphql-engine.git
-  graphql-engine = self.callPackage ../misc/haskell/hasura/graphql-engine {};
+  # hasura graphql-engine is not released to hackage.
+  # https://github.com/hasura/graphql-engine/issues/7391
+  ci-info = self.callPackage ../misc/haskell/hasura/ci-info.nix {};
+  pg-client = self.callPackage ../misc/haskell/hasura/pg-client.nix {
+    resource-pool = self.callPackage ../misc/haskell/hasura/pool.nix {};
+  };
+  graphql-parser = self.callPackage ../misc/haskell/hasura/graphql-parser.nix {};
+  graphql-engine = self.callPackage ../misc/haskell/hasura/graphql-engine.nix {};
```

`graphql-engine` also depends on `resource-pool` and there is a clashing one on hackage unfortunately. Maybe a comment warning about this fact here may be appropriate.

```suggestion
  graphql-engine = self.callPackage ../misc/haskell/hasura/graphql-engine.nix {
      resource-pool = self.callPackage ../misc/haskell/hasura/pool.nix {};
  };
```

---

the pool.nix is actually a fork of the upstream resource-pool.

---

In any case we should use it since it is propagated from `pg-client` — conflicting packages tend to cause problems.

---

yep, testing it now, will push afterwards